### PR TITLE
Keep the original dtype of the data when stretching

### PR DESCRIPTION
**Pull request by pnuu**
The stretching forces the dtype to `float64`. This PR makes sure the original dtype is preserved.

 - [ ] Closes #xxxx (remove if there is no corresponding issue, which should only be the case for minor changes)
 - [x] Tests added (for all bug fixes or enhancements)
 - [ ] Tests passed (for all non-documentation changes)
 - [ ] Passes ``git diff origin/master **/*py | flake8 --diff`` (remove if you did not edit any Python files)
 - [ ] Fully documented (remove if this change should not be visible to users, e.g., if it is an internal clean-up, or if this is part of a larger project that will be documented later)

### CODE REVIEW

**Review by mraspaud — APPROVED** [review]
LGTM! thanks for tackling this issue!

**Review by djhoese — APPROVED** [review]

**Comment by pnuu** [comment]
This latest change to the tests made me think: do we allow integer data? If yes, should I check for that and then use `float32` (or `float64`?) dtype for scale factor?

**Comment by djhoese** [comment]
Integer data in any part of Satpy? Or a specific part? I expect all category products to be integers. And palletized images are also integers I think.

**Comment by pnuu** [comment]
Let me rephrase: are any integer data expected to be passed to `trollimage.XRImage.stretch()`?

**Comment by gerritholl** [comment]
I think integer data can be stretched.  For example, suppose I have data that are discrete (categorical or not) and range from 0 to 10, that I want to write as a greyscale image in which the shades of grey can be seen.  Then I'd want to stretch it to cover 0 to 255.  In that case `palettize` is probably more suitable, but I don't think `stretch` is necessarily invalid.

**Comment by pnuu** [comment]
Thanks @gerritholl I'll see if @djhoese or @mraspaud have thoughs on this before starting anything.

**Comment by djhoese** [comment]
Restarting failed readthedocs job as it appeared to be a download SSL certificate error.

**Review by djhoese — APPROVED** [review]

File: trollimage/tests/test_image.py [hunk 1]
```diff
@@ -1467,19 +1467,34 @@ def test_crude_stretch(self):
...
+        expected_blue = (arr[:, :, 2] - 2.) / (74. - 2.)
+        np.testing.assert_allclose(blue, expected_blue.astype(np.float32), rtol=1e-6)
+
+    def test_crude_stretch_with_limits(self):
         arr = np.arange(75).reshape(5, 5, 3).astype(float)
         data = xr.DataArray(arr.copy(), dims=['y', 'x', 'bands'],
                             coords={'bands': ['R', 'G', 'B']})
         img = xrimage.XRImage(data)
         img.crude_stretch(0, 74)
+        assert img.data.dtype == float
```
Comment (djhoese) [inline trollimage/tests/test_image.py:1486]: I'm not sure how important of a check this is unless it is switched to 32-bit float, but :man_shrugging: not a big deal

Comment (pnuu) [reply trollimage/tests/test_image.py:1486]: This is there just to make sure everything is **not** changed to `float32` ("keep the original dtype").
